### DISCUSSION TITLE
ESIID Mappings

### DISCUSSION THREAD

**Discussion by nealbgit** [ercot/api-specs#61] [General]
To map AMS Interval data UIDESIIDs to ESIIDs, I've been manually using the Market Data Transparency Web Services, and downloading the CSV from "ESIIDs I own for a Trade Date."  I'd prefer to automate the entire process, but there seems to be no way to access the ESIID data programmatically via APIs, etc.  Does anyone have info on how that can be done?

**Comment by DMJ-CV-91913** [comment]
usually that data is private and would be through a secure protocol between QSE/REP and with ERCOT.

In my experience this is all done electronically, EDI (XML structure), though it would not surprise me if it were done via spreadsheet if requested.  That will not be public.   Usually this is reserved for Market Participants.

You'll need an approved certificate  to access MIS.

**Reply by nealbgit** [reply]
Yes DMJ - I have an approved certificate.  

But currently I have to manually go to:

https://mis.ercot.com/secure/applications
![image](https://github.com/user-attachments/assets/ccbd66ac-e707-4923-9ea2-0e50a6bb962c)
Then select 'Provide ESIIDs I own for a trade date', then wait until the file is generated, then click on the link.
![image](https://github.com/user-attachments/assets/0ecd8d02-6930-4484-9832-5ec10b0f3aee)

I'm looking for a way to download this information programmatically.

**Reply by DMJ-CV-91913** [reply]
Probably would need to check with ERCOT.   It seems they may assign a file id, you make a GET request, then based on your certificate, it will initiate a download.  if you find documentation, that would help start the ball rolling.   I cannot access the link you provided above.

ERCOT I believe is undergoing some technical changes.  Some public information is accessible through APIs, this type of information may or may not be.  It is an inconsistent interface structure, but one that takes security seriously.

**Comment by nealbgit** [comment]
@ajay-ercot or @sscott-ercot - do you have any info on how this CSV can be downloaded via API?

**Comment by sjinright-ercot** [comment]
There isn't an API available to specifically pull the Market Data Transparency related data.  It is possible to use the Get Reports service that is exposed as part of the EWS API services to retrieve the download URL for a Report ID (assuming you have a digital certificate that allows it).  It would return something like this:
<ns1:Report>
                  <ns1:operatingDate>2024-12-20</ns1:operatingDate>
                  <ns1:reportGroup>Market Data Transparency Output Files</ns1:reportGroup>
                  <ns1:fileName>ext.00001007.0001234567891000.20241220.092713956.5574202-WS-IDR.zip</ns1:fileName>
                  <ns1:created>2024-12-20T09:27:13</ns1:created>
                  <ns1:size>2651</ns1:size>
                  <ns1:format>zip</ns1:format>
                  <ns1:URL>https://misapitest.ercot.com/misdownload/servlets/mirDownload?doclookupId=123456</ns1:URL>
               </ns1:Report>
  More information can be found here if this service would be helpful.
https://developer.ercot.com/applications/ews/Report%20Messages/GetReports/

**Reply by Ri-Hong-Base** [reply]
Hello, could you provide a link to the Get Reports service endpoint in the repo?

**Reply by sjinright-ercot** [reply]
Hi, 
The specification to Get Reports can be found here:
https://developer.ercot.com/applications/ews/ews/ 
and more specifically here: 
https://developer.ercot.com/applications/ews/Report%20Messages/GetReports/

The WSDL and XSDs can be found here: 
https://github.com/ercot/api-specs/tree/main/ews

Thanks,
Susan

**Comment by haqureshi** [comment]
How can i call Find Transaction Method of Retail API 
(https://misapi.ercot.com/retailAPI/)

Please help
